xds/internal/resolver/xds_resolver_test.go

```diff
@@ -1178,295 +1265,349 @@ func (s) TestXDSResolverMaxStreamDuration(t *testing.T) {
 	if err != nil {
-		t.Fatalf("Error waiting for UpdateState to be called: %v", err)
+		t.Fatalf("Timeout waiting for an update from the resolver: %v", err)
 	}
-	rState = gotState.(resolver.State)
+	rState = val.(resolver.State)
 	if err := rState.ServiceConfig.Err; err != nil {
-		t.Fatalf("ClientConn.UpdateState received error in service config: %v", rState.ServiceConfig.Err)
+		t.Fatalf("Received error in service config: %v", rState.ServiceConfig.Err)
 	}
-	wantJSON2 := `{"loadBalancingConfig":[{
-	 "xds_cluster_manager_experimental":{
-	   "children":{
-		 "cluster:test-cluster-1":{
-		   "childPolicy":[{"cds_experimental":{"cluster":"test-cluster-1"}}]
-		 },
-		 "cluster:NEW":{
-		   "childPolicy":[{"cds_experimental":{"cluster":"NEW"}}]
-		 }
-	   }
-	 }}]}`
-	wantSCParsed2 := internal.ParseServiceConfig.(func(string) *serviceconfig.ParseResult)(wantJSON2)
-	if !internal.EqualServiceConfigForTesting(rState.ServiceConfig.Config, wantSCParsed2.Config) {
-		t.Errorf("ClientConn.UpdateState received different service config")
+	wantSCParsed = internal.ParseServiceConfig.(func(string) *serviceconfig.ParseResult)(`
+{
+	"loadBalancingConfig": [
+		{
+		  "xds_cluster_manager_experimental": {
+			"children": {
+			  "cluster:OLD-CLUSTER": {
+				"childPolicy": [
+				  {
+					"cds_experimental": {
+					  "cluster": "OLD-CLUSTER"
+					} 
+				  } 
+				] 
+			  }, 
+			  "cluster:NEW-CLUSTER": {
+				"childPolicy": [
+				  {
+					"cds_experimental": {
+					  "cluster": "NEW-CLUSTER"
+					} 
+				  } 
+				] 
+			  } 
+			} 
+		  } 
+		} 
+	  ] 
+}`)
+	if !internal.EqualServiceConfigForTesting(rState.ServiceConfig.Config, wantSCParsed.Config) {
+		t.Errorf("Received unexpected service config")
 		t.Error("got: ", cmp.Diff(nil, rState.ServiceConfig.Config))
-		t.Fatal("want: ", cmp.Diff(nil, wantSCParsed2.Config))
+		t.Fatal("want: ", cmp.Diff(nil, wantSCParsed.Config))
 	}
 
-	// Invoke OnCommitted; should lead to a service config update that deletes
-	// test-cluster-1.
-	res.OnCommitted()
+	cs = iresolver.GetConfigSelector(rState)
+	if cs == nil {
+		t.Fatal("Received nil config selector in update from resolver")
+	}
+	resNew, err := cs.SelectConfig(iresolver.RPCInfo{Context: ctx, Method: "/service/method"})
+	if err != nil {
+		t.Fatalf("cs.SelectConfig(): %v", err)
+	}
+	if cluster := clustermanager.GetPickedClusterForTesting(resNew.Context); cluster != "cluster:NEW-CLUSTER" {
+		t.Fatalf("Picked cluster is %q, want %q", cluster, "cluster:NEW-CLUSTER")
+	}
 
-	xdsC.InvokeWatchRouteConfigCallback("", xdsresource.RouteConfigUpdate{
-		VirtualHosts: []*xdsresource.VirtualHost{
-			{
-				Domains: []string{targetStr},
-				Routes:  []*xdsresource.Route{{Prefix: newStringP(""), WeightedClusters: map[string]xdsresource.WeightedCluster{"NEW": {Weight: 1}}}},
-			},
-		},
-	}, nil)
-	gotState, err = tcc.stateCh.Receive(ctx)
+	// Invoke OnCommitted on the old RPC; should lead to a service config update
+	// that deletes the old cluster.
```

Comment:
", as the old cluster no longer has any pending RPCs"? I don't feel strongly about this though.